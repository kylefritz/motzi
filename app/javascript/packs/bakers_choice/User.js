import React from "react";
import _ from "lodash";

export default function User({
  id: userId,
  firstName,
  lastName,
  pickupDay,
  credits,
  breadsPerWeek,
  onPickBread,
  menu,
}) {
  const handleClick = (itemId) => {
    onPickBread({ userId, itemId });
  };

  return (
    <tr>
      <td>{firstName}</td>
      <td>{lastName}</td>
      <td>{credits}</td>
      <td>{breadsPerWeek}</td>
      <td>{pickupDay}</td>
      <td>
        {menu.items.map(({ id, name }) => (
          <button key={id} onClick={() => handleClick(id)}>
            {name}
          </button>
        ))}
      </td>
    </tr>
  );
}
