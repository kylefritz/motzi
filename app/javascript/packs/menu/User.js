import React, {useState} from 'react'
import BuyCredits from "../buy/App";

export const humanizeBreadsPerWeek = (perWeek) => {
  if (perWeek == 0.5) {
    return "Every other week";
  }
  if (perWeek == 1.0) {
    return "Every week";
  }
  if (perWeek == 2.0) {
    return "Two breads per week";
  }
  if (perWeek == 3.0) {
    return "Three breads per week";
  }
  return `${perWeek} breads per week`;
}

export default function User({ user, onRefreshUser }) {
  const [showBuy, setShowBuy] = useState(false)

  // TODO: add weeks remaining
  // <div className="col">
  //   <h5 className="text-center">Weeks remaining</h5>
  //   <p className="text-center"></p>
  // </div>
  return (
    <>
      <div className="row mt-5">
        <div className="col">
          <h5 className="text-center">Ordering for</h5>
          <div className="text-center mb-2">
            <div>{user.name}</div>
            <div>
              <small className="ml-2">
                <a href="/signout" className="text-nowrap">
                  Not you?
                </a>
              </small>
            </div>
          </div>
        </div>
        <div className="col">
          <h5 className="text-center">Ordering frequency</h5>
          <p className="text-center" title={user.breadsPerWeek}>
            {humanizeBreadsPerWeek(user.breadsPerWeek)}
          </p>
        </div>
        <div className="col">
          <h5 className="text-center">Credits remaining</h5>
          <p className="text-center">
            {user.credits}
            {onRefreshUser && (
              <button
                type="button"
                className="btn btn-sm btn-link text-nowrap"
                style={{ fontSize: "80%" }}
                onClick={() => setShowBuy(!showBuy)}
              >
                Buy more
              </button>
            )}
          </p>
        </div>
        <div className="col">
          <h5 className="text-center">Pickup Day</h5>
          <p className="text-center">
            {user.pickupDay}
          </p>
        </div>
      </div>
      {showBuy && <BuyCredits onComplete={onRefreshUser} user={user} />}
    </>
  );
}
