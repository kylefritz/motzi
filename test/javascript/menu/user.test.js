import React from "react";
require("../configure_enzyme");
import { shallow } from "enzyme";

import User, { humanizeBreadsPerWeek } from "menu/User";

test("snapshot", () => {
  const user = {
    name: "kyle",
    credits: 6,
    firstHalf: true,
    breadsPerWeek: 1.0,
  };
  const wrapper = shallow(<User user={user} />);
  expect(wrapper).toMatchSnapshot();
});

test("humanizeBreadsPerWeek", () => {
  expect(humanizeBreadsPerWeek(0.5)).toEqual("Every other week");
  expect(humanizeBreadsPerWeek(1.0)).toEqual("Every week");
  expect(humanizeBreadsPerWeek(2.0)).toEqual("Two breads per week");
  expect(humanizeBreadsPerWeek(3.0)).toEqual("Three breads per week");
  expect(humanizeBreadsPerWeek(5.3)).toEqual("5.3 breads per week");
});
