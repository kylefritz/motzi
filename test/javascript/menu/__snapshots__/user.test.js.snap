// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<Fragment>
  <div
    className="row mt-5"
  >
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Ordering for
      </h5>
      <div
        className="text-center mb-2"
      >
        <div>
          kyle
        </div>
        <div>
          <small
            className="ml-2"
          >
            <a
              className="text-nowrap"
              href="/signout"
            >
              Not you?
            </a>
          </small>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Ordering frequency
      </h5>
      <p
        className="text-center"
        title={1}
      >
        Every week
      </p>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Credits remaining
      </h5>
      <p
        className="text-center"
      >
        6
      </p>
    </div>
    <div
      className="col"
    >
      <h5
        className="text-center"
      >
        Pickup Day
      </h5>
      <p
        className="text-center"
      >
        Thursday
      </p>
    </div>
  </div>
</Fragment>
`;
